FreeRTOS/Test/CBMC/patches/FreeRTOSConfig.h

```diff
@@ -134,6 +134,106 @@ extern void vAssertCalled( const char * pcFile,
 #define configASSERT( x )    if( ( x ) == 0 ) vAssertCalled( __FILE__, __LINE__ )
 #endif
 
+/* The function that implements FreeRTOS printf style output, and the macro
+ * that maps the configPRINTF() macros to that function. */
+extern void vLoggingPrintf( const char * pcFormat,
+                            ... );
+#define configPRINTF( X )    vLoggingPrintf X
+
+/* Non-format version thread-safe print. */
+#define configPRINT_STRING( X )    vLoggingPrint( X )
+
+/* Application specific definitions follow. **********************************/
+
+/* If configINCLUDE_DEMO_DEBUG_STATS is set to one, then a few basic IP trace
+ * macros are defined to gather some UDP stack statistics that can then be viewed
+ * through the CLI interface. */
+#define configINCLUDE_DEMO_DEBUG_STATS       1
+
+/* The size of the global output buffer that is available for use when there
+ * are multiple command interpreters running at once (for example, one on a UART
+ * and one on TCP/IP).  This is done to prevent an output buffer being defined by
+ * each implementation - which would waste RAM.  In this case, there is only one
+ * command interpreter running, and it has its own local output buffer, so the
+ * global buffer is just set to be one byte long as it is not used and should not
+ * take up unnecessary RAM. */
+#define configCOMMAND_INT_MAX_OUTPUT_SIZE    1
+
+/* Only used when running in the FreeRTOS Windows simulator.  Defines the
+ * priority of the task used to simulate Ethernet interrupts. */
+#define configMAC_ISR_SIMULATOR_PRIORITY     ( configMAX_PRIORITIES - 1 )
+
+/* This demo creates a virtual network connection by accessing the raw Ethernet
+ * or WiFi data to and from a real network connection.  Many computers have more
+ * than one real network port, and configNETWORK_INTERFACE_TO_USE is used to tell
+ * the demo which real port should be used to create the virtual port.  The ports
+ * available are displayed on the console when the application is executed.  For
+ * example, on my development laptop setting configNETWORK_INTERFACE_TO_USE to 4
+ * results in the wired network being used, while setting
+ * configNETWORK_INTERFACE_TO_USE to 2 results in the wireless network being
+ * used. */
+#define configNETWORK_INTERFACE_TO_USE       ( 0L )
+
+/* The address of an echo server that will be used by the two demo echo client
+ * tasks:
+ * http://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/TCP_Echo_Clients.html,
+ * http://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/UDP_Echo_Clients.html. */
+#define configECHO_SERVER_ADDR0              192
```

Comment:
Between line 181 - 226. I don't think these are all necessary in TCP/IP stack proofs. And for some, I don't think fixing the addresses are clean. After all we are trying to verify TCP/IP stack logic. 

(e.g. making sure ARP works with any MAC address is one of the many goals... This seems obvious and no brain, but we want to demonstrate such with [formal methods](https://en.wikipedia.org/wiki/Formal_methods).)

In code during test setup, it'll look somewhat like for example  -- 
```
mac_addr0 = nondet_int8_t()
``` 
or 
```
__CPROVER_assume( mac_addr0 > 0 )
__CPROVER_assume( mac_addr0 <255 )
```
(if to leave out say 0.0.0.0 and 255.255.255.255, or any specific range.)

Proposing to take two steps:
- Remove unnecessary macros between 181-226. 
- After this PR gets merged, relax constraint on the proof inputs.